runtime/coverage.go

```diff
@@ -108,23 +107,23 @@ func NewLocationCoverage(lineHits map[int]int) *LocationCoverage {
 // also exclude locations from coverage collection.
 type CoverageReport struct {
 	// Contains a *LocationCoverage per location.
-	Coverage map[common.Location]*LocationCoverage `json:"-"`
+	Coverage map[string]*LocationCoverage `json:"-"`
 	// Contains an *ast.Program per location.
-	Programs map[common.Location]*ast.Program `json:"-"`
+	Programs map[string]struct{} `json:"-"`
```

Comment:
Note: Here I have entirely removed the `*ast.Program` value, because it was not utilized anywhere. This is reduce allocations, and also allow to properly setup a `CoverageReport` after JSON unmarshal, meaning that we can setup which locations/programs where inspected.